### PR TITLE
Shreyas/user profile page

### PR DESCRIPTION
**Pull request by shreyas-thirumale**
# Pull Request

## Description

Created user profile page with the ability to edit name, phone number, and DOB. Users can also change their password for login.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

### Code Review Preparation
- [X] I have performed a self-review and reviewed the changed files
- [X] My code follows the project's style guidelines and passes linting
- [X] I have added comments in hard-to-understand areas and updated documentation

### Testing (Ignore for the first PR)
- [ ] I have added relevant tests and they pass locally
- [ ] I have tested the changes manually, including edge cases

### Code Quality & Security
- [X] I have formatted my code and removed debugging/unused code
- [X] No TypeScript errors or linting issues
- [X] No sensitive information committed (API keys, passwords, etc.)
- [X] User inputs are validated and sanitized

### Dependencies & Deployment
- [X] Dependencies are documented and package-lock.json is updated if needed
- [X] Changes work with Firebase emulator and environment variables are documentedRetry

## Screenshots (If it is a front end feature screenshot is required)
Screenshot 2026-02-16 at 2 49 40 PM
Screenshot 2026-02-16 at 2 49 45 PM

## Release Notes

* **New Features**
  * Edit your account profile information including first name, last name, date of birth, and phone number with form validation.
  * New password reset interface allows you to securely change your password.
  * Account information now displays in a centralized card layout with loading indicators.

* **Chores**
  * Updated dependency versions.

### CODE REVIEW

File: lib/queries/users.ts [hunk 1]
```diff
@@ -70,3 +74,65 @@ function useAllAccounts(onlyActive: boolean) {
...
+        try {
+            const promise = updateMutation.mutateAsync(newUserData);
+            toast.promise(promise, {
+                loading: "Updating account...",
+                success: "Account updated successfully!",
+                error: "Error: Couldn't update account",
+            });
+            await promise;
+        } catch (error) {
+            // Error already handled by toast.promise
```
Comment (constrictingsnake) [inline lib/queries/users.ts]: same issue as profile/page.tsx

File: lib/queries/users.ts [hunk 2]
```diff
@@ -70,3 +74,65 @@ function useAllAccounts(onlyActive: boolean) {
...
+        },
+        enabled: !!uid,
+    }); 
+
+    const updateMutation = useMutation({
+        mutationFn: async (newUserData: UserData) => {
+            await updateUser(newUserData);
+        },
+        onMutate: async (newUserData: UserData) => {
+            await queryClient.cancelQueries({ queryKey: ["user", uid] });
```
Comment (joelchem) [inline lib/queries/users.ts]: For this cancel query you should cancel all queries to any "user" query key. not just for this specific uid

File: app/profile/page.tsx [hunk 3]
```diff
@@ -3,18 +3,372 @@
...
+                            </h1>
+                            <p>Loading...</p>
+                        </div>
+                    </div>
+                </div>
+            </ProtectedRoute>
+        );
+    }
+
+    if (!account) {
```
Comment (joelchem) [inline app/profile/page.tsx]: Same thing about not repeating the whole body in a guard clause as previous comment

File: app/profile/page.tsx [hunk 4]
```diff
@@ -3,18 +3,372 @@
...
+            const firebaseError = error as { code?: string };
+            if (
+                firebaseError.code === "auth/wrong-password" ||
+                firebaseError.code === "auth/invalid-credential"
+            ) {
+                toast.error("Current password is incorrect");
+            } else if (firebaseError.code === "auth/weak-password") {
+                toast.error("New password is too weak");
+            } else {
+                toast.error("Error updating password");
```
Comment (joelchem) [inline app/profile/page.tsx]: In the final error clause here, rethrow the error! We do not want to be catching unhandled errors and not propagating them!!

**Review by constrictingsnake — CHANGES_REQUESTED** [review]

**Review by joelchem — CHANGES_REQUESTED** [review]
Good work overall! The UI looks great and just like the Figma! Some random misc changes. Let us know on Slack if you have any questions about any comments!

File: app/profile/page.tsx [hunk 5]
```diff
@@ -3,18 +3,372 @@
...
+                                                type="password"
+                                                value={confirmPassword}
+                                                onChange={(e) =>
+                                                    setConfirmPassword(
+                                                        e.target.value,
+                                                    )
+                                                }
+                                            />
+                                        </div>
+
```
Comment (joelchem) [inline app/profile/page.tsx]: Can you also include a Red logout button at the bottom left of the page/box. like on the same level as where save and cancel are, but all the way on the left

**Review by joelchem — APPROVED** [review]
